js/viewer/viewer.js

```diff
@@ -45,38 +45,39 @@ var odfViewer = {
 		return (odfViewer.supportedMimes.indexOf(mimetype) !== -1);
 	},
 
-	register : function(){
-		var i,
-		    mime;
-
-		for (i = 0; i < odfViewer.supportedMimes.length; ++i) {
-			mime = odfViewer.supportedMimes[i];
-			OCA.Files.fileActions.registerAction(
-				{
-					name: "Richdocuments",
-					actionHandler: odfViewer.onEdit,
+	register : function() {
+		for (var i = 0; i < odfViewer.supportedMimes.length; ++i) {
+			var mime = odfViewer.supportedMimes[i];
+
+			if(!$("#isPublic").val() &&
```

Comment:
Either the naming is terrible wrong, or the condition is wrong. I assume that, in a public link, this condition should be true, and in a public link I guess we want to prioritize the secure view (if the rest of the conditions apply)